storage_service/locations/models/package.py

```diff
@@ -1019,13 +1028,27 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
 
         # Copy pointer file if exists
         # TODO what do if LOCKSS & uncompressed?
-        if self.package_type in (Package.AIP, Package.AIC):
+        if self.package_type in (Package.AIP, Package.AIC) and to_be_compressed:
             reingest_pointer_src = os.path.join(origin_location.relative_path, os.path.dirname(origin_path), 'pointer.xml')
-            reingest_pointer_name = 'pointer.' + self.uuid + '.reingest.xml'
-            reingest_pointer_dst = os.path.join(ss_internal.relative_path, reingest_pointer_name)
-            src_space.move_to_storage_service(reingest_pointer_src, reingest_pointer_name, internal_space)
-            internal_space.move_from_storage_service(reingest_pointer_name, reingest_pointer_dst)
-            reingest_pointer = os.path.join(ss_internal.full_path, reingest_pointer_name)
+
+            # If reingesting a previously compressed AIP, make a temporary "reingest" pointer (otherwise make a normal one)
+            if was_compressed:
+                reingest_pointer_name = 'pointer.' + self.uuid + '.reingest.xml'
+            else:
+                reingest_pointer_name = 'pointer.' + self.uuid + '.xml'
+
+            reingest_pointer_dst = os.path.join(ss_internal.relative_path, utils.uuid_to_path(self.uuid), reingest_pointer_name)
+
+            try:
+                src_space.move_to_storage_service(reingest_pointer_src, reingest_pointer_name, internal_space)
+                internal_space.move_from_storage_service(reingest_pointer_name, reingest_pointer_dst)
+
+                self.pointer_file_location = Location.active.get(purpose=Location.STORAGE_SERVICE_INTERNAL)
+                self.pointer_file_path = os.path.join(utils.uuid_to_path(self.uuid), 'pointer.{}.xml'.format(self.uuid))
+
+                reingest_pointer = os.path.join(ss_internal.full_path, utils.uuid_to_path(self.uuid), reingest_pointer_name)
+            except:
+                pass
```

Comment:
Please log an error or warning.  `self.pointer_file_location` and `self.pointer_file_path` may also need to be updated.